airflow/api_connexion/parameters.py

```diff
@@ -86,3 +87,32 @@ def wrapped_function(*args, **kwargs):
         return cast(T, wrapped_function)
 
     return format_parameters_decorator
+
+
+def apply_sorting(model, query, order_by, to_replace=None, allowed_attrs=None):
+    """Apply sorting to query"""
+    lstriped_orderby = order_by.lstrip('-')
+    if allowed_attrs and lstriped_orderby not in allowed_attrs:
+        modelname = model.__tablename__.capitalize()
+        model_mapping = {
+            "Ab_user": 'User',
+            "Slot_pool": "Pool",
+            "Dag_run": "DagRun",
+            "Dag": "DagModel",
+            "Ab_role": "Role",
+            "Import_error": "ImportError",
+        }
+        if model_mapping.get(modelname, None):
+            modelname = model_mapping[modelname]
```

Comment:
```suggestion
        modelname = model_mapping.get(modelname, modelname)
```

It feels weird to me the field name replacements are declared outside of this function, but model name replacements inside. Maybe it’d be a good idea to somehow put them together? Maybe a few seperate classes (enums?) to hold those definitions. Not entirely sure.

Reply:
The field names were declared outside this function because it's different for different models while the model name replacements are for a few models.

Reply:
Thinking of having these on the models itself? What do you think?

Reply:
Yeah, probably not a good idea to put them on models directly. I’m thinking a separate definition like how Flask-WTForm classes are separate to the ORM model (it doesn’t even need to be backed by a model).

```python
class ConnectionFilter:
    replacements = {"connection_id": "conn_id"}
    allowed_attrs = {'connection_id', 'conn_type', 'description', 'host', 'port', 'id'}

    @classmethod
    def order_by(cls, query: Query, order_by: str) -> Query:
        ...

def get_connections(session, limit, offset=0, order_by="id"):
    ...
    query = ConnectionFilter.order_by(session.query(Connection), order_by)
    ...
```

Would this feel like an overkill?

Reply:
I feel we can do this later when refactoring, in the process, we can reduce the repetition of model name

Reply:
Sounds reasonable to me. Maybe we should open a follow-up issue for this so we don’t forget?

Reply:
Issue created https://github.com/apache/airflow/issues/15072